server/hydrogen-render/3-render-hydrogen-to-string-unsafe.js

```diff
@@ -42,6 +40,36 @@ async function _renderHydrogenToStringUnsafe({ fromTimestamp, roomData, events,
     };
   }
 
+  const vmContext = vm.createContext(dom);
+  // Make the dom properties available in sub-`require(...)` calls
+  vmContext.global.window = dom.window;
+  vmContext.global.document = dom.document;
+  vmContext.global.Node = dom.Node;
+  vmContext.global.navigator = dom.navigator;
+  vmContext.global.DOMParser = dom.DOMParser;
+  // Make sure `webcrypto` exists since it was only introduced in Node.js v17
+  assert(crypto.webcrypto);
+  vmContext.global.crypto = crypto.webcrypto;
+
+  // So require(...) works in the vm
+  vmContext.global.require = require;
+  // So we can see logs from the underlying vm
+  vmContext.global.console = console;
+
+  return {
+    dom,
+    vmContext,
+  };
+}
```

Comment:
This is just re-arranging things in a function. Just some refactor when I had more changes with `fake-indexeddb` and mock `LocalStoage` which were removed from this PR in https://github.com/matrix-org/matrix-public-archive/pull/43/commits/74b096c1fcdd9ba7889d49d96ea12bb4b9a5c0a0